honeybee/typing.py

```diff
@@ -83,6 +83,42 @@ def float_in_range(value, mi=INFNEG, ma=INFPOS, input_name=''):
     return number
 
 
+def float_in_range_excl(value, mi=INFNEG, ma=INFPOS, input_name=''):
+    """Check a float value to be greater than minimum and less than maximum."""
+    try:
+        number = float(value)
+    except (ValueError, TypeError):
+        raise TypeError('Input {} must be a number. Got {}: {}.'.format(
+            input_name, type(value), value))
```

Comment:
Rather than repeating these same 5 lines of code 6 times, can we just pull it out into its own hidden method?  You can just call it `_number_check` and just call it from each of the other places here.